### PR TITLE
docs: Cloud Providers (that support Intel SGX)

### LINKED ISSUE #957: SGX Compatible Cloud / Bare-metal providers
We need to keep a list of cloud providers which are known to work (or, should work) with SGX, for third-parties to deploy various Oasis related projects on.

Currently:

 * OVH support SGX on their bare-metal servers: https://www.ovhcloud.com/en/bare-metal/prices/?use_cases=confidential-computing
 * Azure support SGX: https://learn.microsoft.com/en-us/azure/confidential-computing/virtual-machine-solutions-sgx for their DCsv2, DCsv3, and DCdsv3 VMs
 * GCore support SGX in some locations: https://gcore.com/cloud/intel-sgx
 * PhoenixNAP support SGX: https://phoenixnap.com/kb/intel-sgx
 * Alibaba cloud: https://www.alibabacloud.com/en/product/ebm?_p_lc=1
 * IBM Cloud: https://cloud.ibm.com/docs/vpc?topic=vpc-creating-virtual-servers-with-sgx-vpc
 * Vultr bare-metal: https://www.vultr.com/products/bare-metal/

The list should provide last-updated time, and also solicit recommendations

### PR DESCRIPTION
**Pull request by CedarMist**
Fixes: #957

[PREVIEW](https://deploy-preview-958--oasisprotocol-docs.netlify.app/node/run-your-node/prerequisites/cloud-providers/)

### CODE REVIEW

**Comment by gw0** [comment]
We should mention that as a quick test one should run the Oasis Attestation tool (https://docs.oasis.io/node/run-your-node/prerequisites/set-up-trusted-execution-environment-tee#oasis-attestation-tool) before proceeding.

File: docs/node/run-your-node/prerequisites/cloud-providers.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+| [Gcore](https://gcore.com) | [Bare Metal](https://gcore.com/cloud/bare-metal-servers) | [Computing with Intel SGX](https://gcore.com/cloud/intel-sgx) | 2024-09-25 |
+| [PhoenixNAP](https://phoenixnap.com/) | [Bare Metal Cloud](https://phoenixnap.com/bare-metal-cloud) | [What is Intel SGX and What are the Benefits?](https://phoenixnap.com/kb/intel-sgx) | 2024-09-25 |
+| [Alibaba Cloud](https://www.alibabacloud.com) | [(ECS) Bare Metal Instances](https://www.alibabacloud.com/en/product/ebm) | [Build an SGX confidential computing environment](https://www.alibabacloud.com/help/en/ecs/user-guide/build-an-sgx-encrypted-computing-environment) | 2024-09-25 |
+| [IBM Cloud](https://cloud.ibm.com/) | [Virtual Private Cloud (VPC)](https://www.ibm.com/cloud/vpc) | [Confidential computing with SGX for VPC](https://cloud.ibm.com/docs/vpc?topic=vpc-about-sgx-vpc). | 2024-09-25 |
+| [Vultr](https://www.vultr.com/) | [Bare Metal](https://www.vultr.com/products/bare-metal/) |  [Intel SGX development on Vultr](https://zenlot.medium.com/intel-sgx-development-on-vultr-30cdfd5c9754) | 2024-09-25 |
```
Comment (matevz) [inline docs/node/run-your-node/prerequisites/cloud-providers.md]: I would order those in alphabetic order so we don't "prefer" anyone over another.

File: docs/node/run-your-node/prerequisites/cloud-providers.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Cloud Providers
+
+Before committing to a service be sure to verify the processor compatibility and enquire with the provider about the status of Intel SGX support. Intel maintains a comprehensive list of processors that support Intel SGX:
```
Comment (matevz) [inline docs/node/run-your-node/prerequisites/cloud-providers.md]: Please wrap to 80 characters. Also below.

File: docs/node/run-your-node/prerequisites/cloud-providers.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+ * https://www.intel.com/content/www/us/en/architecture-and-technology/software-guard-extensions-processors.html
```
Comment (matevz) [inline docs/node/run-your-node/prerequisites/cloud-providers.md:7]: Please also add this reference under the CPU section in https://deploy-preview-958--oasisprotocol-docs.netlify.app/node/run-your-node/prerequisites/hardware-recommendations#suggested-minimum-configurations too

File: docs/node/run-your-node/prerequisites/cloud-providers.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+
+## Possible Limitations
+
+While many bare-metal or dedicated server providers use Intel processors that support SGX, there are potential limitations:
+
+ 1. **BIOS Configuration:** Some providers may not allow customers to access or modify BIOS settings, which is necessary to enable SGX.
```
Comment (matevz) [inline docs/node/run-your-node/prerequisites/cloud-providers.md]: Use unordered bullets here since the order is not important.

**Review by matevz — APPROVED** [review]
I wonder where should we put this. Either:
- under Run your node -> Prerequisites as a new chapter between the Hardware and Staking requirements or
- as a section at the end of the existing Hardware requirements chapter.

Both would work.

File: docs/node/run-your-node/prerequisites/cloud-providers.md (hunk 4 above)
Comment (gw0) [inline docs/node/run-your-node/prerequisites/cloud-providers.md]: Enabling SGX is insufficient, one has to also set other BIOS settings mentioned in https://docs.oasis.io/node/run-your-node/prerequisites/set-up-trusted-execution-environment-tee#bios-configuration. If I am not mistaken, for example on Alibaba you can not disable hyper-threading and consequently SGX does not work.

Comment (CedarMist) [reply docs/node/run-your-node/prerequisites/cloud-providers.md]: https://www.alibabacloud.com/help/en/ecs/user-guide/build-an-sgx-encrypted-computing-environment

Alibaba officially support Intel SGX, I haven't tested them but I assume that given they have lots of documentation around using SGX then SGX would work.

Comment (CedarMist) [reply docs/node/run-your-node/prerequisites/cloud-providers.md]: Have changed to:

> Some providers may not allow customers to access or modify BIOS settings, which may be necessary to correctly configure Intel SGX.

With the assumption that some providers may have specific SGX instance types which have all the necessary BIOS changes, whereas  others you will have to set one or more BIOS options to correctly configure Intel SGX.

**Comment by CedarMist** [comment]
image

**Pull request merged**